pkg/datasource/opensergo/rule_assembler_aggregator.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 1999-2020 Alibaba Group Holding Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package opensergo
+
+import (
+	"github.com/alibaba/sentinel-golang/core/flow"
+	"github.com/alibaba/sentinel-golang/logging"
+	"github.com/opensergo/opensergo-go/pkg/configkind"
+	faulttolerancePb "github.com/opensergo/opensergo-go/pkg/proto/fault_tolerance/v1"
+	"github.com/pkg/errors"
+)
+
+type RuleAssemblerAggregator struct {
+}
+
+func (assembler RuleAssemblerAggregator) assembleFlowRulesFromRateLimitStrategies(pbRules []faulttolerancePb.FaultToleranceRule, pbRlStrategyMap map[string]faulttolerancePb.RateLimitStrategy) []flow.Rule {
```

Comment:
Why not use pointers here? (`*faulttolerancePb.FaultToleranceRule` and `*faulttolerancePb.RateLimitStrategy`)

Reply:
changed